// evergreen-ui/commonjs/themes/classic/components/badge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _tinycolor = _interopRequireDefault(require("tinycolor2"));

var _themeTools = require("../../../theme/src/theme-tools");

var baseStyle = {
  height: 16,
  paddingY: 0,
  paddingX: 6,
  borderRadius: 'radii.1',
  fontSize: '11.5px',
  textAlign: 'center',
  textDecoration: 'none',
  textTransform: 'uppercase'
};

function getContrastingColor(theme, props) {
  var scheme = (0, _themeTools.get)(theme, "fills.".concat(props.appearance, ".").concat(props.color), {
    backgroundColor: props.color,
    color: (0, _tinycolor["default"])(props.color).isLight() ? 'colors.dark' : 'white'
  });
  return {
    color: scheme.color,
    backgroundColor: scheme.backgroundColor
  };
}

var appearances = {
  solid: getContrastingColor,
  subtle: getContrastingColor
};
var sizes = {};
var _default = {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
exports["default"] = _default;
