// evergreen-ui/commonjs/themes/classic/components/button.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _helpers = require("../deprecated/helpers");

var _shared = require("../deprecated/shared");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var baseStyle = {
  fontFamily: 'fontFamilies.ui',
  borderRadius: 'radii.1',
  fontWeight: 500,
  border: '0',
  color: function color(theme, _ref) {
    var _color = _ref.color;
    return theme.colors[_color] || _color || 'colors.default';
  },
  _disabled: _objectSpread({}, _shared.defaultControlStyles.disabled)
};
var appearances = {
  primary: {
    backgroundColor: 'white',
    backgroundImage: function backgroundImage(_, props) {
      return (0, _helpers.getPrimaryButtonStylesForIntent)(props.intent).linearGradient.base;
    },
    boxShadow: function boxShadow(theme) {
      return "inset 0 0 0 1px ".concat(theme.scales.neutral.N5A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N2A);
    },
    color: 'white',
    _focus: {
      backgroundImage: function backgroundImage(_, props) {
        return (0, _helpers.getPrimaryButtonStylesForIntent)(props.intent).linearGradient.focus;
      },
      boxShadow: function boxShadow(theme, props) {
        return "0 0 0 3px ".concat((0, _helpers.getPrimaryButtonStylesForIntent)(props.intent).focusColor, ", inset 0 0 0 1px ").concat(theme.scales.neutral.N4A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N5A);
      }
    },
    _hover: {
      backgroundImage: function backgroundImage(_, props) {
        return (0, _helpers.getPrimaryButtonStylesForIntent)(props.intent).linearGradient.hover;
      }
    },
    _active: {
      backgroundImage: function backgroundImage(_, props) {
        return (0, _helpers.getPrimaryButtonStylesForIntent)(props.intent).linearGradient.active;
      }
    },
    _focusAndActive: {}
  },
  "default": _objectSpread(_objectSpread({}, _shared.defaultControlStyles.base), {}, {
    color: function color(_, props) {
      return (0, _helpers.getTextColorForIntent)(props.intent);
    },
    _hover: _shared.defaultControlStyles.hover,
    _active: _shared.defaultControlStyles.active,
    _focus: _shared.defaultControlStyles.focus,
    _disabled: _shared.defaultControlStyles.disabled
  }),
  minimal: {
    color: function color(theme, props) {
      return (0, _helpers.getTextColorForIntent)(props.intent, theme.colors.blue.base);
    },
    background: 'transparent',
    _hover: {
      background: 'scales.neutral.N2A'
    },
    _focus: {
      boxShadow: function boxShadow(theme) {
        return "0 0 0 3px ".concat(theme.colors.blueAlpha.B5A);
      }
    },
    _active: {
      background: 'scales.blue.B3A'
    }
  }
};
var sizes = {
  small: {
    height: 24,
    minWidth: 24,
    fontSize: 'fontSizes.1',
    lineHeight: '24px',
    paddingLeft: 12,
    paddingRight: 12
  },
  medium: {
    height: 32,
    minWidth: 32,
    fontSize: 'fontSizes.1',
    lineHeight: '32px',
    paddingLeft: 16,
    paddingRight: 16
  },
  large: {
    height: 40,
    minWidth: 40,
    fontSize: 'fontSizes.2',
    lineHeight: '40px',
    paddingLeft: 20,
    paddingRight: 20
  }
};
var _default = {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
exports["default"] = _default;
