// evergreen-ui/commonjs/themes/classic/components/input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var baseStyle = {
  borderRadius: 'radii.1',
  lineHeight: 'lineHeights.1',
  color: 'colors.default',
  fontSize: 'fontSizes.1',
  paddingX: 10,
  _placeholder: {
    color: 'colors.neutralAlpha.N6A'
  },
  _disabled: {
    cursor: 'not-allowed'
  },
  _focus: {
    zIndex: 'zIndices.focused'
  }
};
var appearances = {
  "default": {
    backgroundColor: 'white',
    boxShadow: function boxShadow(theme) {
      return "inset 0 0 0 1px ".concat(theme.colors.neutralAlpha.N5A, ", inset 0 1px 2px ").concat(theme.colors.neutralAlpha.N4A);
    },
    _invalid: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.colors.red.base, ", inset 0 1px 2px ").concat(theme.colors.neutralAlpha.N4A);
      }
    },
    _focus: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 2px ".concat(theme.colors.neutralAlpha.N4A, ", inset 0 0 0 1px ").concat(theme.colors.blue.B7, ", ").concat(theme.shadows.focusRing);
      }
    },
    _disabled: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.colors.neutralAlpha.N4A);
      },
      backgroundColor: 'colors.neutralAlpha.N2A'
    }
  },
  neutral: {
    backgroundColor: 'colors.neutralAlpha.N2A',
    _invalid: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.colors.red.base);
      }
    },
    _focus: {
      backgroundColor: 'white',
      boxShadow: function boxShadow(theme) {
        return "0 0 0 2px ".concat(theme.colors.blueAlpha.B6A);
      }
    },
    _disabled: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.colors.neutralAlpha.N4A);
      }
    }
  },
  none: {
    backgroundColor: 'white',
    _disabled: {
      backgroundColor: 'colors.neutral.N2'
    }
  }
};
var sizes = {
  small: {
    height: 24
  },
  medium: {
    height: 32
  },
  large: {
    height: 40,
    lineHeight: 'lineHeights.2'
  }
};
var _default = {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
exports["default"] = _default;
