// evergreen-ui/commonjs/themes/classic/components/switch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var baseStyle = {};
var appearances = {
  "default": {
    _base: {
      color: 'white',
      backgroundColor: 'scales.neutral.N5A'
    },
    _disabled: {
      opacity: 0.5,
      backgroundImage: 'none'
    },
    _hover: {
      backgroundColor: 'scales.neutral.N5A'
    },
    _active: {
      backgroundColor: 'scales.neutral.N6A'
    },
    _focus: {
      boxShadow: function boxShadow(theme) {
        return "0 0 0 3px ".concat(theme.scales.blue.B6A);
      }
    },
    _checked: {
      backgroundColor: 'scales.blue.B8',
      color: 'white'
    },
    _checkedHover: {
      backgroundColor: 'scales.blue.B8',
      color: 'white'
    },
    _checkedActive: {
      backgroundColor: 'scales.blue.B9',
      color: 'white'
    },
    _checkedDisabled: {}
  }
};
var sizes = {};
var _default = {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
exports["default"] = _default;
