// evergreen-ui/commonjs/nudge/src/Nudge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Nudge = exports.Pulsar = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _glamor = require("glamor");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Position = _interopRequireDefault(require("../../constants/src/Position"));

var _layers = require("../../layers");

var _scales = require("../../scales");

var _theme = require("../../theme");

var _tooltip = require("../../tooltip");

var _POSITION_KEYS;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var pulseAnimation = _glamor.css.keyframes('pulseAnimation', {
  '0%': {
    transform: 'scale(1)'
  },
  '50%': {
    transform: 'scale(1.9)'
  },
  '100%': {
    transform: 'scale(1)'
  }
});

var animationTiming = 'cubic-bezier(0, 0, 0.58, 1)';
var animationDuration = '1.8s';
var pulsarAnimationClassName = (0, _glamor.css)({
  animation: "".concat(pulseAnimation, " ").concat(animationDuration, " ").concat(animationTiming, " both infinite")
}).toString();
var POSITION_KEYS = (_POSITION_KEYS = {}, (0, _defineProperty2["default"])(_POSITION_KEYS, _Position["default"].TOP_LEFT, ['top', 'left']), (0, _defineProperty2["default"])(_POSITION_KEYS, _Position["default"].TOP_RIGHT, ['top', 'right']), (0, _defineProperty2["default"])(_POSITION_KEYS, _Position["default"].BOTTOM_LEFT, ['bottom', 'left']), (0, _defineProperty2["default"])(_POSITION_KEYS, _Position["default"].BOTTOM_RIGHT, ['bottom', 'right']), _POSITION_KEYS);

var getPositionProps = function getPositionProps(_ref) {
  var position = _ref.position,
      size = _ref.size;
  var keys = POSITION_KEYS[position];
  var props = {};
  keys.forEach(function (key) {
    var isYAxisKey = key === 'top' || key === 'bottom';

    if (isYAxisKey) {
      props[key] = -(size / 2);
    } else {
      props[key] = -size;
    }
  });
  return props;
};

var Pulsar = /*#__PURE__*/(0, _react.memo)(function (_ref2) {
  var _ref2$position = _ref2.position,
      position = _ref2$position === void 0 ? _Position["default"].TOP_RIGHT : _ref2$position,
      _ref2$size = _ref2.size,
      size = _ref2$size === void 0 ? (0, _scales.majorScale)(1) : _ref2$size;

  var _useTheme = (0, _theme.useTheme)(),
      colors = _useTheme.colors;

  var positionProps = getPositionProps({
    position: position,
    size: size
  });
  var outerPadding = size * 0.25;
  return /*#__PURE__*/_react["default"].createElement(_layers.Pane, (0, _extends2["default"])({
    position: "absolute",
    borderRadius: "50%",
    backgroundColor: colors.blue100,
    boxSizing: "content-box",
    opacity: 0.7,
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    padding: outerPadding,
    className: pulsarAnimationClassName
  }, positionProps), /*#__PURE__*/_react["default"].createElement(_layers.Pane, {
    width: size,
    height: size,
    backgroundColor: colors.blue200,
    borderRadius: "50%",
    opacity: 0.7
  }));
});
exports.Pulsar = Pulsar;
Pulsar.propTypes = {
  /**
   * The position of the pulsar
   */
  position: _propTypes["default"].oneOf([_Position["default"].TOP_LEFT, _Position["default"].TOP_RIGHT, _Position["default"].BOTTOM_LEFT, _Position["default"].BOTTOM_RIGHT]),

  /**
   * The width/height of the dot
   */
  size: _propTypes["default"].number
};

var Nudge = function Nudge(_ref3) {
  var children = _ref3.children,
      _ref3$isShown = _ref3.isShown,
      isShown = _ref3$isShown === void 0 ? false : _ref3$isShown,
      _ref3$position = _ref3.position,
      position = _ref3$position === void 0 ? _Position["default"].TOP_RIGHT : _ref3$position,
      size = _ref3.size,
      tooltipContent = _ref3.tooltipContent;
  return /*#__PURE__*/_react["default"].createElement(_tooltip.Tooltip, {
    content: tooltipContent,
    position: position,
    isShown: isShown ? undefined : false
  }, /*#__PURE__*/_react["default"].createElement(_layers.Pane, {
    position: "relative"
  }, isShown && /*#__PURE__*/_react["default"].createElement(Pulsar, {
    position: position,
    size: size
  }), children));
};

exports.Nudge = Nudge;
Nudge.displayName = "Nudge";
Nudge.propTypes = {
  /**
   * The position for the Puslar and the Tooltip
   */
  position: _propTypes["default"].oneOf([_Position["default"].TOP_LEFT, _Position["default"].TOP_RIGHT, _Position["default"].BOTTOM_LEFT, _Position["default"].BOTTOM_RIGHT]),

  /**
   * Size of the Pulsar
   */
  size: Pulsar.propTypes.size,

  /**
   * Wether or not the Pulsar/Tooltip is shown
   */
  isShown: _propTypes["default"].bool,

  /**
   * Content for the tooltip
   */
  tooltipContent: _tooltip.Tooltip.propTypes.content,

  /**
   * Content for the pulsar/tooltip to be anchored too
   */
  children: _propTypes["default"].node
};
