// evergreen-ui/esm/themes/classic/tokens/fills.js
import colors from './colors';
/**
 * Used for matching background colors and foreground colors
 */

var fills = {
  solid: {
    neutral: {
      color: 'white',
      backgroundColor: colors.neutral.base
    },
    blue: {
      color: 'white',
      backgroundColor: colors.blue.base
    },
    red: {
      color: 'white',
      backgroundColor: colors.red.base
    },
    orange: {
      color: 'white',
      backgroundColor: colors.orange.base
    },
    yellow: {
      color: colors.yellow.dark,
      backgroundColor: colors.yellow.base
    },
    green: {
      color: 'white',
      backgroundColor: colors.green.base
    },
    teal: {
      color: 'white',
      backgroundColor: colors.teal.base
    },
    purple: {
      color: 'white',
      backgroundColor: colors.purple.base
    }
  },
  subtle: {
    neutral: {
      color: colors.neutral.dark,
      backgroundColor: colors.neutral.light
    },
    blue: {
      color: colors.blue.dark,
      backgroundColor: colors.blue.light
    },
    red: {
      color: colors.red.dark,
      backgroundColor: colors.red.light
    },
    orange: {
      color: colors.orange.dark,
      backgroundColor: colors.orange.light
    },
    yellow: {
      color: colors.yellow.dark,
      backgroundColor: colors.yellow.light
    },
    green: {
      color: colors.green.dark,
      backgroundColor: colors.green.light
    },
    teal: {
      color: colors.teal.dark,
      backgroundColor: colors.teal.light
    },
    purple: {
      color: colors.purple.dark,
      backgroundColor: colors.purple.light
    }
  }
};
export default fills;
