// evergreen-ui/esm/themes/classic/components/table-row.js
var colorMap = {
  none: {
    base: 'white',
    hover: 'colors.neutral.lightest',
    focus: 'colors.neutral.lightest',
    active: 'colors.blue.lightest',
    current: 'colors.blue.lightest'
  },
  danger: {
    base: 'colors.red.lightest',
    hover: 'colors.red.lightest',
    focus: 'colors.red.light',
    active: 'colors.red.light',
    current: 'colors.red.light'
  },
  warning: {
    base: 'colors.yellow.lightest',
    hover: 'colors.yellow.lightest',
    focus: 'colors.yellow.light',
    active: 'colors.yellow.light',
    current: 'colors.yellow.light'
  },
  success: {
    base: 'colors.green.lightest',
    hover: 'colors.green.lightest',
    focus: 'colors.green.light',
    active: 'colors.green.light',
    current: 'colors.green.light'
  }
};

var getBackgroundForIntentAndState = function getBackgroundForIntentAndState(intent, state) {
  return colorMap[intent][state];
};

var baseStyle = {
  outline: 'none',
  textDecoration: 'none',
  height: 48,
  _isSelectable: {
    cursor: 'pointer'
  }
};
var appearances = {
  "default": {
    backgroundColor: function backgroundColor(_, props) {
      return getBackgroundForIntentAndState(props.intent, 'base');
    },
    _hover: {
      backgroundColor: function backgroundColor(_, props) {
        return getBackgroundForIntentAndState(props.intent, 'hover');
      }
    },
    _focus: {
      backgroundColor: function backgroundColor(_, props) {
        return getBackgroundForIntentAndState(props.intent, 'focus');
      }
    },
    _active: {
      backgroundColor: function backgroundColor(_, props) {
        return getBackgroundForIntentAndState(props.intent, 'active');
      }
    },
    _current: {
      backgroundColor: function backgroundColor(_, props) {
        return getBackgroundForIntentAndState(props.intent, 'current');
      }
    }
  }
};
export default {
  baseStyle: baseStyle,
  appearances: appearances
};
