// evergreen-ui/commonjs/themes/classic/deprecated/foundational-styles/fills.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _colors = _interopRequireDefault(require("./colors"));

var _palette = _interopRequireDefault(require("./palette"));

/**
 * Fills are used in Avatars and Badges.
 */
var fills = {};
fills.solid = {
  neutral: {
    color: 'white',
    backgroundColor: _palette["default"].neutral.base
  },
  blue: {
    color: 'white',
    backgroundColor: _palette["default"].blue.base
  },
  red: {
    color: 'white',
    backgroundColor: _palette["default"].red.base
  },
  orange: {
    color: 'white',
    backgroundColor: _palette["default"].orange.base
  },
  yellow: {
    color: _palette["default"].yellow.dark,
    backgroundColor: _palette["default"].yellow.base
  },
  green: {
    color: 'white',
    backgroundColor: _palette["default"].green.base
  },
  teal: {
    color: 'white',
    backgroundColor: _palette["default"].teal.base
  },
  purple: {
    color: 'white',
    backgroundColor: _palette["default"].purple.base
  }
};
fills.subtle = {
  neutral: {
    color: _colors["default"].text["default"],
    backgroundColor: _palette["default"].neutral.light
  },
  blue: {
    color: _palette["default"].blue.dark,
    backgroundColor: _palette["default"].blue.light
  },
  red: {
    color: _palette["default"].red.dark,
    backgroundColor: _palette["default"].red.light
  },
  orange: {
    color: _palette["default"].orange.dark,
    backgroundColor: _palette["default"].orange.light
  },
  yellow: {
    color: _palette["default"].yellow.dark,
    backgroundColor: _palette["default"].yellow.light
  },
  green: {
    color: _palette["default"].green.dark,
    backgroundColor: _palette["default"].green.light
  },
  teal: {
    color: _palette["default"].teal.dark,
    backgroundColor: _palette["default"].teal.light
  },
  purple: {
    color: _palette["default"].purple.dark,
    backgroundColor: _palette["default"].purple.light
  }
};
fills.options = Object.keys(fills.solid);
var _default = fills;
exports["default"] = _default;
