// evergreen-ui/commonjs/themes/classic/deprecated/foundational-styles/palette.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _scales = _interopRequireDefault(require("./scales"));

var _default = {
  neutral: {
    lightest: _scales["default"].neutral.N1,
    light: _scales["default"].neutral.N4,
    base: _scales["default"].neutral.N9,
    dark: _scales["default"].neutral.N10
  },
  blue: {
    lightest: _scales["default"].blue.B1,
    light: _scales["default"].blue.B4,
    base: _scales["default"].blue.B9,
    dark: _scales["default"].blue.B10
  },
  red: {
    lightest: '#fef6f6',
    light: '#fae2e2',
    base: '#ec4c47',
    dark: '#bf0e08'
  },
  orange: {
    lightest: '#fdf8f3',
    light: '#fae3cd',
    base: '#d9822b',
    dark: '#95591e'
  },
  yellow: {
    lightest: '#fef8e7',
    light: '#fbe6a2',
    base: '#f7d154',
    dark: '#7e6514'
  },
  green: {
    lightest: '#f1faf5',
    light: '#d4eee2',
    base: '#47b881',
    dark: '#00783e'
  },
  teal: {
    lightest: '#f1fbfc',
    light: '#d2eef3',
    base: '#14b5d0',
    dark: '#007489'
  },
  purple: {
    lightest: '#f8f7fc',
    light: '#eae7f8',
    base: '#735dd0',
    dark: '#37248f'
  }
};
exports["default"] = _default;
