// evergreen-ui/esm/form-field/src/FormFieldValidationMessage.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["children"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { memo, forwardRef } from 'react';
import { ErrorIcon } from '../../icons';
import { Pane } from '../../layers';
import { Paragraph } from '../../typography';
var FormFieldValidationMessage = /*#__PURE__*/memo( /*#__PURE__*/forwardRef(function FormFieldValidationMessage(_ref, ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Pane, _extends({
    ref: ref,
    display: "flex"
  }, props), /*#__PURE__*/React.createElement(ErrorIcon, {
    color: "danger",
    marginTop: 1,
    size: 14,
    marginRight: 8
  }), /*#__PURE__*/React.createElement(Paragraph, {
    marginTop: 0,
    size: 300,
    color: "danger",
    role: "alert"
  }, children));
}));
FormFieldValidationMessage.propTypes = _objectSpread({}, Pane.propTypes);
export default FormFieldValidationMessage;
