// evergreen-ui/commonjs/themes/classic/tokens/fills.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _colors = _interopRequireDefault(require("./colors"));

/**
 * Used for matching background colors and foreground colors
 */
var fills = {
  solid: {
    neutral: {
      color: 'white',
      backgroundColor: _colors["default"].neutral.base
    },
    blue: {
      color: 'white',
      backgroundColor: _colors["default"].blue.base
    },
    red: {
      color: 'white',
      backgroundColor: _colors["default"].red.base
    },
    orange: {
      color: 'white',
      backgroundColor: _colors["default"].orange.base
    },
    yellow: {
      color: _colors["default"].yellow.dark,
      backgroundColor: _colors["default"].yellow.base
    },
    green: {
      color: 'white',
      backgroundColor: _colors["default"].green.base
    },
    teal: {
      color: 'white',
      backgroundColor: _colors["default"].teal.base
    },
    purple: {
      color: 'white',
      backgroundColor: _colors["default"].purple.base
    }
  },
  subtle: {
    neutral: {
      color: _colors["default"].neutral.dark,
      backgroundColor: _colors["default"].neutral.light
    },
    blue: {
      color: _colors["default"].blue.dark,
      backgroundColor: _colors["default"].blue.light
    },
    red: {
      color: _colors["default"].red.dark,
      backgroundColor: _colors["default"].red.light
    },
    orange: {
      color: _colors["default"].orange.dark,
      backgroundColor: _colors["default"].orange.light
    },
    yellow: {
      color: _colors["default"].yellow.dark,
      backgroundColor: _colors["default"].yellow.light
    },
    green: {
      color: _colors["default"].green.dark,
      backgroundColor: _colors["default"].green.light
    },
    teal: {
      color: _colors["default"].teal.dark,
      backgroundColor: _colors["default"].teal.light
    },
    purple: {
      color: _colors["default"].purple.dark,
      backgroundColor: _colors["default"].purple.light
    }
  }
};
var _default = fills;
exports["default"] = _default;
